src/controllers/FacetSliderQueryController.ts

```diff
@@ -307,4 +271,85 @@ export class FacetSliderQueryController {
   private getBrowserCompatibleFormat(value: string) {
     return value.replace('@', 'T').replace(/\//g, '-');
   }
+
+  private putGroupByForFullRangeSliderIntoQueryBuilder(queryBuilder: QueryBuilder, basicGroupByRequest: IGroupByRequest) {
+    // We need a group by request for the "full range" that does not take into account the current query (query override)
+    // The goal is to obtain a query wich return the whole range of results without taking into account what the users typed in the search box,
+    // so that the X range of the slider is static between each queries.
+
+    this.groupByRequestForFullRange = queryBuilder.groupByRequests.length;
+    const groupByRequestForFullRange = clone(basicGroupByRequest);
+
+    // This removes the "basic" query override. ie: user input
+    delete groupByRequestForFullRange.queryOverride;
+    // This removes any current filter in the facet caused by a selection
+    delete groupByRequestForFullRange.advancedQueryOverride;
+    // We keep the "constantQueryOverride", since it normally contains static filter always applied no matter what.
+
+    if (this.facet.options.queryOverride) {
+      groupByRequestForFullRange.advancedQueryOverride = this.facet.options.queryOverride;
+    }
+
+    this.addExpressionToPotentiallyRemoveInvalidDocumentDates(groupByRequestForFullRange);
+    queryBuilder.groupByRequests.push(groupByRequestForFullRange);
+  }
+
+  private putGroupByBasicSliderIntoQueryBuilder(queryBuilder: QueryBuilder): IGroupByRequest {
+    let maximumNumberOfValues = 1;
+
+    if (this.facet.hasAGraph()) {
+      maximumNumberOfValues = this.facet.options.graph.steps;
+    }
+
+    let rangeValues = undefined;
+    const { start, end } = this.formatStartAndEnd();
```

Comment:
Instead of destructuring here and then using the values in the object on line 308, what about using the spread operator on the object returned by `this.formatStartAndEnd()` directly in the object? Also, it would be clearer if renamed to `getFormattedStartAndEnd`.

Maybe that method returning more than `start` and `end` properties that we are trying to exclude from the object?